### PR TITLE
Test for recursive validation

### PR DESCRIPTION
**Pull request by mrenvoize**
Add tests for validation of recursive schemata.  Proves the recursive validation works as expected, always good to have more tests to prevent regressions :)

### CODE REVIEW

**Comment by jhthorsen** [comment]
Please use .perltidyrc to format the code.

Other than that: :+1:

**Comment by mrenvoize** [comment]
As requested.. I thought I already had :$

**Comment by jhthorsen** [comment]
I don't think you have: I see four spaces in the code, but... https://github.com/jhthorsen/json-validator/blob/master/.perltidyrc#L6

Would be nice if you rebased this PR and force pushed, but that's not a big deal.

**Comment by mrenvoize** [comment]
Oh, that's odd.. It looks like my pertidy isn't picking up the .perltidyrc file :(

Sorry about that, I'll re-base.

File: t/validate-recursive.t [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+use Test::Mojo;
+use Test::More;
+use JSON::Validator 'validate_json';
+
+{
+  use Mojolicious::Lite;
+  post '/' => sub {
+    my $c = shift;
+    my @errors = validate_json $c->req->json, 'data://main/spec.json';
+    $c->render(status => @errors ? 400 : 200, text => "@errors");
```
Comment (jhthorsen) [inline t/validate-recursive.t]: Isn't it better to do ```json => \@errors``` ? I believe that would contribute to better unit testing than content_like().

**Comment by mrenvoize** [comment]
How about this, both setting json as the response type, and also checking the json error response more explicitly with json_is.

**Comment by jhthorsen** [comment]
Okidoki! :+1:

**Pull request merged**